### PR TITLE
Removing log4 transitive dependencies from bigtable-hbase-1.*

### PR DESCRIPTION
**Pull request by sduskis**
This change has been prompted by a request from a user.  This change was inspired by https://github.com/derjust/cloud-bigtable-client/commit/c775b09b7a9a64344910f17a679afbc42bccfb77

### CODE REVIEW

File: bigtable-hbase-1.0/pom.xml [hunk 1]
```diff
@@ -52,6 +52,10 @@ limitations under the License.
             <groupId>org.apache.hbase</groupId>
             <artifactId>hbase-common</artifactId>
         </dependency>
+        <dependency>
```
Comment (kevinsi4508) [inline bigtable-hbase-1.0/pom.xml]: Will this exclude transient dependencies, or we need to exclude them?
https://github.com/derjust/cloud-bigtable-client/commit/c775b09b7a9a64344910f17a679afbc42bccfb77

Just want to confirm.

Comment (derjust) [inline bigtable-hbase-1.0/pom.xml]: Err no. 
See high level comment above

**Comment by derjust** [comment]
I think that is not working as intendet:

```
mvn dependency:tree |grep log4j
[INFO] |  |  \- log4j:log4j:jar:1.2.17:compile
[INFO] |  |  \- org.slf4j:slf4j-log4j12:jar:1.6.1:compile
[INFO] |  +- log4j:log4j:jar:1.2.17:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.5:compile
[INFO] \- org.slf4j:log4j-over-slf4j:jar:1.7.5:runtime
[INFO] |  +- log4j:log4j:jar:1.2.17:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.5:compile
[INFO] \- org.slf4j:log4j-over-slf4j:jar:1.7.5:runtime
[INFO] |  |  |  \- log4j:log4j:jar:1.2.17:compile
[INFO] |  |  |  \- org.slf4j:slf4j-log4j12:jar:1.6.1:compile
[INFO] |  \- org.slf4j:log4j-over-slf4j:jar:1.7.5:runtime
[INFO] |  +- log4j:log4j:jar:1.2.17:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.5:compile
[INFO] |  \- org.slf4j:log4j-over-slf4j:jar:1.7.5:runtime
[INFO] |  +- log4j:log4j:jar:1.2.17:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.5:compile
```

My approach in derjust@c775b09 was 
* Exclude from the Hbase dependency `slf4j-log4j12` 
* Explicitly also exclude from the Hbase dependency `log4j` as it is a direct `compile` dependency, too
* Add as new dependency `log4j-over-slf4j` as `runtime` dependency to have at runtime log4j API calls to succeed from within the Hbase code

Does that makes sense?

**Comment by sduskis** [comment]
I'm not sure why my top level pom.xml exclusions didn't make it into the PR.  That's where the exclusions ought to be managed to avoid duplication.

**Comment by sduskis** [comment]
I updated the root pom.xml.  It now looks like this:

[INFO] |  +- log4j:log4j:jar:1.2.17:provided
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.5:provided
[INFO] \- org.slf4j:log4j-over-slf4j:jar:1.7.5:runtime
[INFO] |  +- log4j:log4j:jar:1.2.17:provided
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.5:provided
[INFO] \- org.slf4j:log4j-over-slf4j:jar:1.7.5:runtime
[INFO] |  \- org.slf4j:log4j-over-slf4j:jar:1.7.5:runtime
[INFO] |  +- log4j:log4j:jar:1.2.17:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.5:compile
[INFO] |  \- org.slf4j:log4j-over-slf4j:jar:1.7.5:runtime
[INFO] |  +- log4j:log4j:jar:1.2.17:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.5:compile

File: pom.xml [hunk 2]
```diff
@@ -367,6 +374,14 @@ limitations under the License.
                         <groupId>com.google.guava</groupId>
                         <artifactId>guava</artifactId>
                     </exclusion>
+                    <exclusion>
```
Comment (kevinsi4508) [inline pom.xml]: exclude in hadoop-common too?

**Comment by AngusDavis** [comment]
Is there any way to ensure that these don't make it back in via the enforcer plugin or something similar?

**Comment by derjust** [comment]
LGTM 
Opening an issue to add an enforcer rule http://maven.apache.org/enforcer/enforcer-rules/bannedDependencies.html?

**Comment by sduskis** [comment]
I was adding the enforcer rule and the hadoop-common exclusion before the comment.  I checked it in.  PTAL.

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**